testing/xfpga/test_properties_c.cpp

```diff
@@ -45,61 +46,193 @@ class properties_p1 : public ::testing::TestWithParam<std::string> {
+    ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &accel_filter_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(accel_filter_, FPGA_ACCELERATOR), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaEnumerate(&accel_filter_, 1, accel_tokens_.data(), accel_tokens_.size(),
+                            &num_matches_), FPGA_OK);
+    EXPECT_EQ(num_matches_, platform_.devices.size());
+    ASSERT_EQ(xfpga_fpgaOpen(accel_tokens_[0], &accel_, 0), FPGA_OK);
+    ASSERT_EQ(fpgaClearProperties(accel_filter_), FPGA_OK);
+
+    ASSERT_EQ(xfpga_fpgaGetProperties(nullptr, &dev_filter_), FPGA_OK);
+    ASSERT_EQ(fpgaPropertiesSetObjectType(dev_filter_, FPGA_DEVICE), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaEnumerate(&dev_filter_, 1, dev_tokens_.data(), dev_tokens_.size(),
+                            &num_matches_), FPGA_OK);
     EXPECT_EQ(num_matches_, platform_.devices.size());
-    ASSERT_EQ(xfpga_fpgaOpen(tokens_[0], &accel_, 0), FPGA_OK);
-    ASSERT_EQ(fpgaClearProperties(filter_), FPGA_OK);
+    ASSERT_EQ(xfpga_fpgaOpen(dev_tokens_[0], &dev_, 0), FPGA_OK);
+    ASSERT_EQ(fpgaClearProperties(dev_filter_), FPGA_OK);
+
     num_matches_ = 0xc01a;
     invalid_device_ = test_device::unknown();
   }
 
   virtual void TearDown() override {
-    EXPECT_EQ(fpgaDestroyProperties(&filter_), FPGA_OK);
-    EXPECT_EQ(fpgaDestroyProperties(&props_), FPGA_OK);
+    EXPECT_EQ(fpgaDestroyProperties(&prop_), FPGA_OK);
+    EXPECT_EQ(fpgaDestroyProperties(&accel_filter_), FPGA_OK);
+    EXPECT_EQ(fpgaDestroyProperties(&dev_filter_), FPGA_OK);
     EXPECT_EQ(xfpga_fpgaClose(accel_), FPGA_OK);
-    for (auto &t : tokens_) {
+    EXPECT_EQ(xfpga_fpgaClose(dev_), FPGA_OK);
+
+    for (auto &t : accel_tokens_) {
+      if (t) {
+        EXPECT_EQ(xfpga_fpgaDestroyToken(&t), FPGA_OK);
+        t = nullptr;
+      }
+    }
+
+    for (auto &t : dev_tokens_) {
       if (t) {
         EXPECT_EQ(xfpga_fpgaDestroyToken(&t), FPGA_OK);
         t = nullptr;
       }
     }
+
     system_->finalize();
   }
 
-  std::array<fpga_token, 2> tokens_;
-  fpga_properties filter_;
-  fpga_properties props_;
-  fpga_handle handle_;
+  std::array<fpga_token, 2> accel_tokens_;
+  std::array<fpga_token, 2> dev_tokens_;
+  fpga_properties prop_;
+  fpga_properties accel_filter_;
+  fpga_properties dev_filter_;
   fpga_handle accel_;
+  fpga_handle dev_;
   uint32_t num_matches_;
   test_platform platform_;
   test_device invalid_device_;
   test_system* system_;
 };
 
-TEST_P(properties_p1, from_handle) {
-  EXPECT_EQ(xfpga_fpgaGetPropertiesFromHandle(accel_, &props_), FPGA_OK);
+/**
+ * @test    from_handle
+ * @brief   Tests: xfpga_fpgaGetPropertiesFromHandle
+ * @details Given a null fpga_properties* object<br>
+ *          When I call xfpga_fpgaGetPropertiesFromHandle with an valid handle,
+ *          expected result is FPGA_OK.<br>
+ */
+TEST_P(properties_c_p, from_handle) {
+  EXPECT_EQ(xfpga_fpgaGetPropertiesFromHandle(accel_, &prop_), FPGA_OK);
```

Comment:
prop_ is leaked in all cases where it is passed as a parameter to fpgaGetProperties[FromHandle].